### ISSUE TITLE
Tuple/List type default value parsing error

### ISSUE THREAD

**Issue by wu-clan** [DanCardin/cappa#106]
Look this MRE:
```py
import dataclasses

from typing import Annotated

import cappa


@cappa.command
@dataclasses.dataclass
class Example:
    version: Annotated[bool, cappa.Arg(short='-V', long=True, default=False)]
    start_project: Annotated[tuple[str, str], cappa.Arg(short='-s', default=('httpfpt_project', '.'), required=False)]
    start_project2: Annotated[list[str], cappa.Arg(short='-s2', default=['', ''],required=False, num_args=2)]
    end_project: Annotated[int, cappa.Arg(short='-e', default=1, required=False)]


if __name__ == '__main__':
    a = cappa.parse(Example)
    print(a)
```
```shell
# Version: 0.17.0

❯ python .\s.py -s 
Example(version=False, start_project=(), start_project2=['', ''], end_project=1)

❯ python .\s.py -s2
Example(version=False, start_project=('2',), start_project2=['', ''], end_project=1)

# After commenting the start_project
❯ python .\s.py -s2
Example(version=False, start_project2=['[]'], end_project=1)
```

**Comment by pawamoy** [comment]
I believe short options can only be of length 1 (a single character). So in your example `-s2` is parsed as `-s` then `2`.

Try using something else for your example, like `-t`.

**Comment by wu-clan** [comment]
Hi, @pawamoy 

I don't think the short option can only be one character, if you've used a short option with more than one character.

This could be an inadvertent bug.

When I changed `-s2` to `-t`:

```shell
❯ python .\s.py -s
Example(version=False, start_project=(), start_project2=['', ''], end_project=1)

❯ python .\s.py -t
Example(version=False, start_project=('httpfpt_project', '.'), start_project2=['[]'], end_project=1)

❯ python .\s.py -t
Example(version=False, start_project2=['[]'], end_project=1)
```

I think there is still a default value parsing error.

Unless I actively enter parameters, otherwise, it should always output the default value.

**Comment by pawamoy** [comment]
Sure, there might be other bugs, but at least using `-t` in the example makes it less ambiguous as to what could be the cause.

Now, I'm not sure defaults are supposed to work like this. Defaults are when you don't provide the flag/option. Here you provide it, but you don't provide additional positional arguments. Maybe this logic should live in your own code. Not sure though, sorry for the noise if I'm wrong, and I'll let @DanCardin chime in :smile:

**Comment by DanCardin** [comment]
So I think definitely a few issues being raised here.

1. `tuple[str, str]` ought to indicate num_args=2 and fail parsing if you only give it 1, i think.
2. `["[]"]` is not resulting from a default so much as something wrong is going on and it's getting parsed as a string somewhere. `-t 2 3` yields `["[2, 3]"]`
3. currently `short=` truncates the short string to the first character. I suppose there's no technical reason why this couldn't just be the whole string...so long as argparse acts the same way as the cappa parser, i'm happy to remove this

tl;dr seems to be 2 inference bugs and a feature request?

**Comment by pawamoy** [comment]
It looks like argparse accepts short options with more than one characters, and then prevents using the `-x0` syntax where `x` is the short option and `0` is the value:

```python
import argparse
parser = argparse.ArgumentParser()
parser.add_argument("-st")
parser.add_argument("-t")

print(parser.parse_args())
```

```console
% python arg.py -t 1 -st 2
Namespace(st='2', t='1')
% python arg.py -t1 -st 2
Namespace(st='2', t='1')
% python arg.py -t1 -st2
arg.py: error: unrecognized arguments: -st2
```

Yet I wouldn't obviously recommend supporting this as this can create ambiguous parsers:

```python
import argparse
parser = argparse.ArgumentParser()
parser.add_argument("-st")
parser.add_argument("-t")
parser.add_argument("-s", nargs=argparse.ZERO_OR_MORE)

print(parser.parse_args())
```

```console
% python arg.py -st1 
Namespace(st=None, t=None, s=['t1'])
% python arg.py -st 1
Namespace(st='1', t=None, s=None)
```

**Comment by DanCardin** [comment]
I...dont know if i'd consider it ambiguous, but i guess argparse's behavior is more surprising with `-st1 ` than what i'd have expected, so I suppose it is ambiguous :P

**Comment by DanCardin** [comment]
Feel free to check out `dc/inference-handling` branch. it does not allow multicharacter `short` names. But I think it does fix the other oddities you uncovered.

Since this does extend (and solify more inference behavior "in stone"), any input on whether what i point out in the PR are logical would be appreciated.

**Comment by wu-clan** [comment]
I tested this branch on a local build.

1. Input parameters are not required for tuple type and are resolved to `()`.
```shell
❯ python .\s.py -s  
Example(version=False, start_project=(), start_project2=['', ''], end_project=1)
```
2. Short options confused, I think this is still a logical error.
```shell
❯ python .\s.py -ss
Example(version=False, start_project=('s',), start_project2=['', ''], end_project=1)

❯ python .\s.py -sss
Example(version=False, start_project=('ss',), start_project2=['', ''], end_project=1)

❯ python .\s.py -t  
Example(version=False, start_project=('httpfpt_project', '.'), start_project2=['[]'], end_project=1)

❯ python .\s.py -tt
Example(version=False, start_project=('httpfpt_project', '.'), start_project2=["['t']"], end_project=1)

❯ python .\s.py -ttt
Example(version=False, start_project=('httpfpt_project', '.'), start_project2=["['tt']"], end_project=1)
```

3. The long option seems to have been broken.
```shell
# When i set -ttt to start_project2 long option
❯ python .\s.py -ttt 
Usage: example [-V] [-s START_PROJECT] [---ttt START_PROJECT2] [-e END_PROJECT] [-h] [--completion COMPLETION]

Error: Unrecognized arguments: -ttt
```

**Comment by wu-clan** [comment]
Sorry, I don't have any extra operations, it's all based on my first MRE and `dc/inference-handling` branch.

**Comment by pawamoy** [comment]
> # When i set -ttt to start_project2 long option

Do you mean you set `long="-ttt"`? That's why I was asking for your declarations.
Values for `long` must look like `--optname` or `optname`, not `-optname`.

**Comment by DanCardin** [comment]
On the `dc-inference-handling` branch, and your MRE I get:
```
❮ python mre.py -s
Usage: example [-V] [-s START_PROJECT START_PROJECT] [-s2 START_PROJECT2
START_PROJECT2] [-e END_PROJECT] [-h] [--completion COMPLETION]

Error: Argument '-s' requires 2 values, found 0
```

You original mre has the non-working `short="-s2"`, so adjusting that to `-t`, I get equivalent behavior there:
```
❯ python meow.py -ttt
Usage: example [-V] [-s START_PROJECT START_PROJECT] [-t START_PROJECT2
START_PROJECT2] [-e END_PROJECT] [-h] [--completion COMPLETION]

Error: Argument '-t' requires 2 values, found 1 ('tt' so far)
```

Makes me think you may not actually be on the branch?
---

But pawamoy seems right based on the error message, `long='-ttt'` as a way around multicharacter short options not being allowed isn't going to work how you want, they're parsed differently.

Disallowing multicharacter short options really **is** an artificial limitation, but it'd also be rather easy to create short args which cannot be targeted (i.e. your original `-s` and `-s2` example) based on his above example of how argparse will interpret the values.

**Comment by wu-clan** [comment]
Sorry about the noise.

I added a new environment to test the dc branch. 

Here is the MRE and the test：
```py
import dataclasses

from typing import Annotated

import cappa


@cappa.command
@dataclasses.dataclass
class Example:
    version: Annotated[bool, cappa.Arg(short='-V', long=True, default=False)]
    start_project: Annotated[tuple[str, str], cappa.Arg(short='-s', default=('httpfpt_project', '.'), required=False)]
    start_project2: Annotated[list[str], cappa.Arg(short='-t', long='--ttt', default=['', ''],required=False, num_args=2)]
    end_project: Annotated[int, cappa.Arg(short='-e', default=1, required=False)]


if __name__ == '__main__':
    a = cappa.parse(Example)
    print(a)
```
```shell
❯ python .\s.py   
Example(version=False, start_project=('httpfpt_project', '.'), start_project2=['', ''], end_project=1)

❯ python .\s.py -s
Usage: example [-V] [-s START_PROJECT START_PROJECT] [-t START_PROJECT2 START_PROJECT2] [-e END_PROJECT] [-h] [--completion COMPLETION]

Error: Argument '-s' requires 2 values, found 0

❯ python .\s.py -ss
Usage: example [-V] [-s START_PROJECT START_PROJECT] [-t START_PROJECT2 START_PROJECT2] [-e END_PROJECT] [-h] [--completion COMPLETION]

Error: Argument '-s' requires 2 values, found 1 ('s' so far)

❯ python .\s.py -t  
Usage: example [-V] [-s START_PROJECT START_PROJECT] [-t START_PROJECT2 START_PROJECT2] [-e END_PROJECT] [-h] [--completion COMPLETION]

Error: Argument '-t/--ttt' requires 2 values, found 0

❯ python .\s.py -tt
Usage: example [-V] [-s START_PROJECT START_PROJECT] [-t START_PROJECT2 START_PROJECT2] [-e END_PROJECT] [-h] [--completion COMPLETION]

Error: Argument '-t/--ttt' requires 2 values, found 1 ('t' so far)

❯ python .\s.py --ttt 
Usage: example [-V] [-s START_PROJECT START_PROJECT] [-t START_PROJECT2 START_PROJECT2] [-e END_PROJECT] [-h] [--completion COMPLETION]

Error: Argument '-t/--ttt' requires 2 values, found 0
```

Everything looks fine.

At the same time, I tested `cappa.argparse.backend`, and their behavior was consistent.
 
Unless I miss something, I think the current problem has been fixed.

Thank you for youz quick response.
